bf2fc6cc711aee1a0c2a/kas-fleet-manager
# feat: add kafka ibp version information metrics

## Description

Related to https://issues.redhat.com/browse/MGDSTRM-6360.

This PR adds Kafka IBP Prometheus metrics information.


## Verification Steps

* Run the tests
* Checkout the branch and build the component. Make sure you have a kafka running and ready and you can go to the metrics endpoint and you should see something like this:

```
msoriano@localhost:~$ curl 127.0.0.1:8080/metrics | grep -i version  | grep -i kafka
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 28914    0 28914    0     0  1660k      0 --:--:-- --:--:-- --:--:-- 1764k
# HELP kafka_ibp_version Reports the version of Kafka in terms of seconds since the epoch.\nThe type 'actual' is the Kafka IBP version that is reported by kas-fleetshard.\nThe type 'desired' is the desired Kafka IBP version that is set in the kas-fleet-manager.\nIf the type is 'upgrade' it means the Kafka IBP version is being upgraded.\n
# TYPE kafka_ibp_version gauge
kafka_ibp_version{cluster_id="1okjbuf3n8um779l4vjdlbj4dn16mp0f",kafka_id="c6egeuf170houcu5f880",type="actual",version="2.8"} 1.637681407e+09
kafka_ibp_version{cluster_id="1okjbuf3n8um779l4vjdlbj4dn16mp0f",kafka_id="c6egeuf170houcu5f880",type="desired",version="2.8"} 1.637681407e+09
# HELP kafka_version Reports the version of Kafka in terms of seconds since the epoch.\nThe type 'actual' is the Kafka version that is reported by kas-fleetshard.\nThe type 'desired' is the desired Kafka version that is set in the kas-fleet-manager. \nIf the type is 'upgrade' it means the Kafka is being upgraded.\n
# TYPE kafka_version gauge
kafka_version{cluster_id="1okjbuf3n8um779l4vjdlbj4dn16mp0f",kafka_id="c6egeuf170houcu5f880",type="actual",version="2.8.1"} 1.637681407e+09
kafka_version{cluster_id="1okjbuf3n8um779l4vjdlbj4dn16mp0f",kafka_id="c6egeuf170houcu5f880",type="desired",version="2.8.1"} 1.637681407e+09
strimzi_version{cluster_id="1okjbuf3n8um779l4vjdlbj4dn16mp0f",kafka_id="c6egeuf170houcu5f880",type="actual",version="strimzi-cluster-operator.v0.23.0-5"} 1.637681407e+09
strimzi_version{cluster_id="1okjbuf3n8um779l4vjdlbj4dn16mp0f",kafka_id="c6egeuf170houcu5f880",type="desired",version="strimzi-cluster-operator.v0.23.0-5"} 1.637681407e+09
```

## Checklist (Definition of Done)
